File: example-webapp/src/main/java/vip/yazilim/libs/springvip/template/example/Person.java
package vip.yazilim.libs.springvip.template.example;

import javax.persistence.*;

import lombok.Data;

@Entity
@Table(name = "person")
@Data // just for lombok. Not required!!
public class Person {



    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(columnDefinition = "serial")
    private int id;


    @Column(name = "first_name")
    private String firstName;

    @Column(name = "last_name")
    private String lastName;

    private String career;
}

File: src/main/kotlin/vip/yazilim/springvip/templates/kotlin/entity/Person.kt
package vip.yazilim.springvip.templates.kotlin.entity

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.validation.constraints.NotBlank

/**
 *
 * @author maemresen - maemresen@yazilim.vip
 * 12/30/2020
 */
@Entity
data class Person(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    val id: Long,

    @NotBlank(message = "name is a mandatory name")
    var name: String,

    @NotBlank(message = "age is a mandatory field")
    var age: Double
)
